tests/unit/authenticators/oauth2-implicit-grant-test.js

```diff
@@ -0,0 +1,121 @@
+/* jshint expr:true */
+/* jscs:disable requireDotNotation */
+import Ember from 'ember';
+import { it } from 'ember-mocha';
+import { describe, beforeEach, afterEach } from 'mocha';
+import { expect } from 'chai';
+import Pretender from 'pretender';
+import OAuth2ImplicitGrant from 'ember-simple-auth/authenticators/oauth2-implicit-grant';
+
+const { tryInvoke } = Ember;
+
+describe('OAuth2ImplicitGrantAuthenticator', () => {
+  let authenticator;
+  let server;
+
+  const accessToken = {
+    'access_token': 'secret-token',
+    'state': 'abcd',
+    'token_type': 'bearer'
+  };
+
+  const invalidTokenErr = {
+    'error': 'invalid_token',
+    'error_description': 'The received token is invalid'
+  };
+
+  const accessDeniedErr = {
+    'error': 'access_denied',
+    'error_description': 'Access denied'
+  };
+
+  beforeEach(() => {
+    authenticator = OAuth2ImplicitGrant.create();
+    server        = new Pretender();
+  });
+
+  afterEach(() => {
+    tryInvoke(server, 'shutdown');
+  });
+
+  describe('#restore', () => {
+    describe('when the data contains an access_token', () => {
+      it('resolves with the correct data', (done) => {
+        authenticator.restore(accessToken).then((_data) => {
+          expect(_data).to.eql(accessToken);
+          done();
+        });
```

Comment:
When testing promise based behavior you can return the promise from the test instead of using `done`, e.g.:

```js
describe('when the data does not contain an access_token', () => {
  it('resolves with the correct data', () => {
    return authenticator.restore(accessToken).then((_data) => {
      expect(_data).to.eql(accessToken);
    });
  });
```